packages/x-data-grid/src/hooks/features/rowSelection/gridRowSelectionSelector.ts

```diff
@@ -1,27 +1,42 @@
 import { createSelector, createSelectorMemoized } from '../../../utils/createSelector';
-import { gridRowsLookupSelector } from '../rows/gridRowsSelector';
+import { gridDataRowIdsSelector, gridRowsLookupSelector } from '../rows/gridRowsSelector';
 import { GridStateCommunity } from '../../../models/gridStateCommunity';
 import { GridRowId, GridRowModel } from '../../../models/gridRows';
+import { gridFilteredRowCountSelector } from '../filter/gridFilterSelector';
 
 export const gridRowSelectionStateSelector = (state: GridStateCommunity) => state.rowSelection;
 
 export const selectedGridRowsCountSelector = createSelector(
   gridRowSelectionStateSelector,
-  (selection) => selection.length,
+  gridFilteredRowCountSelector,
+  (selection, filteredRowCount) => {
+    if (selection.type === 'include') {
+      return selection.ids.size;
+    }
+    // TODO: is this correct?
```

Comment:
as far as I can see, it will count in rows that are not selectable but are not in the exclusion list

Reply:
The exclude selection model is only used when all rows are selectable.
Added a comment to mention this.

Anything else you can think of that won't be covered here?

Reply:
In that case, I think that it should be fine